_posts/2020-06-30-Making-IrishGen-SPARQL-Part-II.md

```diff
@@ -0,0 +1,252 @@
+}
+```
+
+The visualisation that is returned by Stardog Studio is below.
+
+<img src="{{site.baseurl}}/assets/images/construct_baeth_1.png" />
+
+As one can see, this produces a star-like structure of children around
+Báeth.  At this point one can start exploring the wider graph by
+right-clicking on the nodes and choosing "Expand from node", which
+will pull from the entire graph (the construct query being just a
+starting point).  However, in Stardog Studio there is a problem where
+unless the "Query All Graphs" property is set to true in Stardog, it
+will not expand.  Briefly, GraphDB has "Query All Graphs" essentially
+on by default and is not a property that the user can set.  Stardog
+has the property off by default which can cause confusion when a user
+is encountering both Triplestores.  A second and more pressing problem
+is that expanding nodes in the face of reasoning can cause a query
+that a user's laptop cannot finish in a reasonable time or finish at
+all in the face of the query timeout restrictions.  In Stardog, this
+can be overcome by turning reasoning off when doing an exploration of
+the graph if the query does not return in a reasonable time but this
+limits what is possible.  It is up to the user's discretion to
+understand what their query involves and what compromises they are
+willing to tolerate.
+
+To set the Stardog parameter "Query All Graphs" to true in Stardog
+Studio.  The user will need to go to "Databases" then select their
+database from the list of databases.  In the right hand pane, the user
+will need to choose "Properties".  In the search box directly below
+the "Properties" header, type "query all".  This will cause the option
+to appear and the user can then check the checkbox to enable "Query
+All Graphs".
+
+To contrast this with a `select` query that returns the same
+information but in a textual format, the query below recreates the
+`construct` query.
+
+```sparql
+prefix rel: <http://purl.org/vocab/relationship/>
+
+select ?child
+from <tag:stardog:api:context:all>
+where {
+    <http://example.com/Rawl_B502/clann_aingeda.trig#Báeth> rel:parentOf ?child
+}
+```
+
+with the result being:
+
+| child                                                    |
+| http://example.com/Rawl_B502/úib_luchta.trig#Comdellach  |
+| http://example.com/Rawl_B502/úib_luchta.trig#Fogartach   |
+| http://example.com/Rawl_B502/úib_luchta.trig#Díummassach |
+| http://example.com/Rawl_B502/úib_luchta.trig#Flann       |
+| http://example.com/Rawl_B502/úib_luchta.trig#Flanngus    |
+| http://example.com/Rawl_B502/úib_luchta.trig#Lassirne    |
+| http://example.com/Rawl_B502/clann_eichlich.trig#Nárgusa |
+| http://example.com/Rawl_B502/úib_luchta.trig#Aingid      |
+| http://example.com/Rawl_B502/úib_luchta.trig#Eichlech    |
+| http://example.com/Rawl_B502/úib_luchta.trig#Airechtach  |
+| http://example.com/Rawl_B502/úib_luchta.trig#Dóelgus     |
+| http://example.com/Rawl_B502/úib_luchta.trig#Abiél       |
```

Comment:
Maybe conclude this with a sentence along the lines of "The same set of URLs have been returned, but merely as a list which cannot be further visualised or queried."